docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,2165 @@
+        }
+
+        $productReviewSummaryTransfer = $this->productReviewSummaryCalculator
+            ->calculate($this->createRatingAggregationTransfer($productReviews));
+
+        $productViewTransfer->setRating($productReviewSummaryTransfer);
+
+        return $productViewTransfer;
+    }
+}
+```
+
+## Model
+
+```
+[Organization]
+├── Zed
+│   └── [Module]
+│       ├── Communication
+│       │   └── [CustomDirectory]
+│       │       ├── [ModelName]Interface.php
+│       │       └── [ModelName].php   
+│       └── Business
+│           └── [CustomDirectory]
+│               ├── [ModelName]Interface.php
+│               └── [ModelName].php  
+├── Yves
+│   └── [Module]
+│       └── [CustomDirectory]
+│           ├── [ModelName]Interface.php
+│           └── [ModelName].php   
+├── Glue
+│   └── [Module]
+│       └── [CustomDirectory]
+│           ├── [ModelName]Interface.php
+│           └── [ModelName].php    
+├── Client
+│   └── [Module]
+│       └── [CustomDirectory]
+│           ├── [ModelName]Interface.php
+│           └── [ModelName].php                
+└── Service
+    └── [Module]
+        └── [CustomDirectory]
+            ├── [ModelName]Interface.php
+            └── [ModelName].php    
+```
+
+**Description**
+
+`Models` encapsulate logic and can be utilized across various layers within the application architecture, within to boundaries of each [layer's specific responsibilities](#layers).
+- [Communication layer](#communication-layer-responsibilities) `Models` are present in [Yves](#yves), [Glue](#glue), and [Client](#client).
+- [Business layer](#business-layer-responsibilities) `Models` are present in [Zed](#zed) and [Service](#service).
+- [Persistence layer](#persistence-layer-responsibilities) `Models` are present in [Zed](#zed).
+
+**Conventions**
+
+- Each model class MUST define and implement an interface (`[ModelName]Interface.php`) that holds the specification of each `public` method.
+- Dependencies MUST be injected via constructor.
+  - `Models` MUST NOT instantiate any object but [Transfer Objects](#transfer-object).
+- Dependencies MUST be referred/used/defined by interface.
+- Dependencies MUST be from the same module, either `Models`, [Repository](#repository), [Entity Manager](#entity-manager), [Config](#module-configurations), or [Bridge wrapped facades](#bridge).
+  - `Models` MUST NOT interact directly with other module's `Models` (eg: via inheritance, shared constants, instantiation, etc.).
+- `Models` MUST be grouped under a folder by activity (eg: `Writers/ProductWriter.php`, `Readers/ProductReader.php`).
+- `Models` MUST NOT be named using generic words, such as `Executor`, `Handler`, or `Worker`.
+- `Model` methods MUST NOT be `private`.
+
+<details><summary markdown='span'>Additional Conventions for Project Development</summary>
+- Dependencies can be referred directly via class.
+- It is recommended to NOT define interface for `Models` without project specific reason.
```

Comment:
``` suggestion
- It is allowed NOT to define interface for `Models` without project specific reason.
```

Reply:
I would leave "recommended" because it makes the codebase thin and when the need of an interface appears it is easy&quick to add it.